plugins/auth-backend-module-guest-provider/src/authenticator.ts

```diff
@@ -15,12 +15,22 @@
  */
 
 import { createProxyAuthenticator } from '@backstage/plugin-auth-node';
+import { NotImplementedError } from '@backstage/errors';
 
 export const guestAuthenticator = createProxyAuthenticator({
   defaultProfileTransform: async () => {
     return { profile: {} };
   },
-  initialize() {},
+  initialize({ config }) {
+    const allowOutsideDev = config.getOptionalBoolean(
+      'backend.auth.dangerouslyAllowOutsideDevelopment',
```

Comment:
This config keys lives in `auth.providers.guest`, but this config object as already scoped to that so you can skip the prefix altogether. We don't necessarily need a test here, but please verify that it's working as intended manually.

Reply:
Of cours, thank you for explaing. I'll fix

Reply:
The error message running with `NODE_ENV=production`: 
```
ForwardedError: Plugin 'auth' startup failed; caused by Error: Failed to initialize guest auth provider, The guest provider cannot be used outside of a development environment
```